### ISSUE TITLE
linter: `eslint/sort-keys/allowLineSeparatedGroups` is not working

### ISSUE THREAD

**Issue by ryansuhartanto** [oxc-project/oxc#15797]
### What version of Oxlint are you using?

1.29.0

### What command did you run?

`oxlint --config linter/eslint/sort-keys/allowLineSeparatedGroups/.oxlintrc.json linter/eslint/sort-keys/allowLineSeparatedGroups`

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "plugins": [
    "eslint"
  ],
  "rules": {
    "eslint/sort-keys": [
      "error",
      {
        "allowLineSeparatedGroups": true
      }
    ]
  }
}
```

### What happened?

Checkout this file or visit this repo <https://github.com/ryansuhartanto/oxc-issues/tree/main/linter/eslint/sort-keys/allowLineSeparatedGroups>:
```ts
interface Meow {
  name: string;
  age: number;
  color: string;
}

export const meow: Meow = {
  name: "Whiskers", // Empty line below 👇️

  age: 3, // Empty line above 👆️
  color: "Tabby",
};
```

It disregard the `allowSeparateTypeImports` option with this error:
```
> oxc-issues@ lint [omitted]/oxc-issues
> oxlint --type-aware --config linter/eslint/sort-keys/allowLineSeparatedGroups/.oxlintrc.json linter/eslint/sort-keys/allowLineSeparatedGroups


  × eslint(sort-keys): Object keys should be sorted
    ╭─[linter/eslint/sort-keys/allowLineSeparatedGroups/index.ts:7:27]
  6 │     
  7 │ ╭─▶ export const meow: Meow = {
  8 │ │     name: "Whiskers",
  9 │ │   
 10 │ │     age: 3,
 11 │ │     color: "Tabby",
 12 │ ╰─▶ };
    ╰────
  help: Replace `name: "Whiskers",
        
          age: 3,
          color: "Tabby"` with `age: 3, color: "Tabby", name: "Whiskers"`.

Found 0 warnings and 1 error.
Finished in 104ms on 1 file with 53 rules using 8 threads.
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by xuanduc987** [comment]
The document for this rule is incorrect, the first option is sort order so the correct rule should be
```
{
    "sort-keys": ["error", "asc", {"allowLineSeparatedGroups": true}]
}
```

**Comment by connorshea** [comment]
Fixed by merging #15904, the docs on the website will be auto-updated with the oxlint release sometime tomorrow. Thanks!
